src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+                commands.spawn((
+                    Connection { id },
+                ));
+                entities.connections.insert(id.clone(), commands.current_entity().expect("`spawn` did not create an entity"));
+
+                // Create a hashmap to hold stream MPSCs for this connection
+                session
+                    .stream_senders
+                    .insert(id.clone(), Default::default());
+            }
+
+            ReceiveEvent::Disconnected(id) => {
+
+                // Delete the entity representing this connection
+                if let Some(e) = entities.connections.remove(&id) {
+                    commands.despawn(e);
+                }
+
+                // drop all stream senders
+                session.stream_senders.remove(&id);
+            }
+
+            // Store the MPSC to send to this stream in the session state
+            ReceiveEvent::OpenedStream {
+                connection_id,
+                stream_id,
+                ref sender,
+            } => {
+                session
+                    .stream_senders
+                    .entry(connection_id.clone())
+                    .or_default()
+                    .insert(stream_id.clone(), sender.clone());
+            }
+
+            // When the socket closes throw away all session state
+            ReceiveEvent::SocketClosed => {
+
+                // Delete all connection entities
+                for (_, e) in entities.connections.drain() {
+                    commands.despawn(e);
+                }
+
+                // drop all stream senders
+                session.stream_senders.clear();
+            }
+
+            _ => {}
+        }
+
+        // Publish event for other systems to consume
+        net_events.send(event);
+    }
+}
+
+// Take ECS events and forward them to MPSCs to be sent over the network
+pub fn send_net_events(mut session: ResMut<SessionEventListenerState>, send_events: ResMut<Events<SendEvent>>) {
+    // Publish packets ready to send to appropriate MPSC channels
+    for send in session.send_event_reader.iter(&send_events) {
+        match send {
+            SendEvent::SendPacket { connection_id, stream_id, data } => {
+                let sender = session
+                    .stream_senders
+                    .entry(connection_id.clone())
+                    .or_default()
+                    .get(&stream_id);
+
+                if let Some(sender) = sender {
+                    if let Err(e) = sender.send(data.to_owned()) {
+                        error!("Send Error: {:?}", e);
+                    }
+                } else {
+                    error!(
+                        "Attempted to send to a non-existant stream: {:?}:{:?}",
+                        connection_id, stream_id
+                    );
+                }
+            }
+        }
+    }
+}
+
+// Handle all the work of a single connection (waiting for new streams to open)
```

Comment:
```suggestion
/// Handle all the work of a single connection (waiting for new streams to open)
```
Might as well turn this into a doc comment since you already did the work to explain it